torch/_refs/__init__.py

```diff
@@ -456,21 +465,37 @@ def _isfinite(a: TensorLikeType) -> TensorLikeType:
 
 
 def _isinf(a: TensorLikeType) -> TensorLikeType:
-    # TODO Add complex tensor support to remove is_infinite prim
-    # if utils.is_complex_dtype(a):
-    #     return bitwise_or(_isinf(real(a), _isinf(imag(a))
-    # else:
-    #     return bitwise_not(bitwise_or(isnan(a), isfinite(a)))
-    if utils.is_float_dtype(a.dtype) or utils.is_complex_dtype(a.dtype):
-        return prims.is_infinite(a)
-
-    return zeros_like(a, dtype=torch.bool)
+    if utils.is_complex_dtype(a.dtype):
+        return bitwise_or(_isinf(real(a)), _isinf(imag(a)))
+    return bitwise_and(bitwise_not(isfinite(a)), bitwise_not(_isnan(a)))
```

Comment:
again -- logical and not bitwise here?

Reply:
There is no `logical_not`, so I keep the `bitwise_not` and added a TODO.